src/d3a/models/strategy/__init__.py

```diff
@@ -454,14 +455,28 @@ def __init__(self):
         self._bids = {}
         self._traded_bids = {}
 
-    def post_bid(self, market, price, energy, buyer_origin=None):
+    def _remove_existing_bids(self, market: Market) -> None:
+        """Remove all existing bids in the market"""
+
+        for bid in self.get_posted_bids(market):
+            assert bid.buyer == self.owner.name
+            if bid.id in market.bids.keys():
+                bid = market.bids[bid.id]
+
+            self.remove_bid_from_pending(market.id, bid.id)
+
+    def post_bid(self, market, price, energy, replace_existing=False, buyer_origin=None):
+        if replace_existing:
+            self._remove_existing_bids(market)
```

Comment:
Unfortunately, this is the one step: Here, in the BaseStrategy, the devices track their bids. Its right to replace them here. However, they also need to be removed from the markets.
So i suggest to get the uuid of the bid here and then delete them from the market (from the external strategy, not from within here).

Reply:
Thanks for the help! 
Isn't this what `self.remove_bid_from_pending(market.id, bid.id)` is doing? It gets the current market, then calls `market.delete_bid(b_id)` for every bid ID.

Reply:
You are indeed right! Sorry, I was too quick with my answer (and thought that there is no market interaction in the BaseStrategy, but there is).
Ok, then all good!